lib/Adapter/TolerantParser/Fixer/MemberEmptyLineFixer.php

```diff
@@ -0,0 +1,174 @@
+
+use Microsoft\PhpParser\Node;
+use Microsoft\PhpParser\Node\ClassConstDeclaration;
+use Microsoft\PhpParser\Node\MethodDeclaration;
+use Microsoft\PhpParser\Node\PropertyDeclaration;
+use Microsoft\PhpParser\Node\TraitUseClause;
+use Microsoft\PhpParser\Parser;
+use Phpactor\CodeBuilder\Domain\StyleProposer;
+use Phpactor\CodeBuilder\Domain\TextEdit;
+use Phpactor\CodeBuilder\Domain\TextEdits;
+use Phpactor\CodeBuilder\Util\TextUtil;
+
+class MemberEmptyLineFixer implements StyleProposer
+{
+    private const META_SUCCESSOR = 'successor';
+    private const META_NODE_CLASS = 'class';
+    private const META_FIRST = 'first';
+    private const META_PRECEDING_BLANK_LINES = 'preceding_blank_lines';
+    private const META_PRECEDING_BLANK_START = 'blank_start';
+    private const META_PRECEDING_BLANK_LENGTH = 'blank_length';
+    private const META_INDENTATION = 'indentation';
+    private const META_IS_METHOD = 'is_method';
+
+    /**
+     * @var Parser
+     */
+    private $parser;
+
+    /**
+     * @var string
+     */
+    private $newLineChar;
+
+    public function __construct(Parser $parser, string $newLineChar = "\n")
+    {
+        $this->parser = $parser;
+        $this->newLineChar = $newLineChar;
+    }
+
+    public function propose(string $text): TextEdits
+    {
+        $node = $this->parser->parseSourceFile($text);
+
+        $membersMeta = $this->gatherMetadata($node, [
+            TraitUseClause::class,
+            ClassConstDeclaration::class,
+            PropertyDeclaration::class,
+            MethodDeclaration::class,
+        ]);
+
+        return TextEdits::fromTextEdits($this->textEdits($membersMeta));
+    }
+
+    private function gatherMetadata(Node $node, array $nodeTypes): array
+    {
+        $nodesMeta = [];
+        $nodes = [];
+
+        foreach ($node->getDescendantNodes() as $decendantNode) {
+            if (!in_array(get_class($decendantNode), $nodeTypes)) {
+                continue;
+            }
+
+            $nodes[] = $decendantNode;
+        }
+
+        $previousNode = null;
+        foreach ($nodes as $node) {
+            assert($node instanceof Node);
+            $meta = [
+                self::META_SUCCESSOR => false,
+                self::META_NODE_CLASS => get_class($node),
+                self::META_FIRST => false,
+                self::META_PRECEDING_BLANK_LINES => $this->countBlankLines($node->getLeadingCommentAndWhitespaceText()),
+                self::META_PRECEDING_BLANK_START => $node->getFullStart(),
+                self::META_PRECEDING_BLANK_LENGTH => $node->getStart() - $node->getFullStart(),
+                self::META_INDENTATION => $this->indentation($node),
+                self::META_IS_METHOD => $node instanceof MethodDeclaration,
+            ];
+
+            if (null === $previousNode) {
+                $meta[self::META_FIRST] = true;
+            }
+
+            if ($previousNode && $previousNode === $meta[self::META_NODE_CLASS]) {
```

Comment:
```suggestion
            if ($previousNode === $meta[self::META_NODE_CLASS]) {
```

I would also rename it to `$previousNodeClass` to be perfectly clear about what it is.